cmd/snap-update-ns/utils.go

```diff
@@ -66,10 +66,26 @@ func (e *ReadOnlyFsError) Error() string {
 	return fmt.Sprintf("cannot operate on read-only filesystem at %s", e.Path)
 }
 
-// Create directories for all but the last segments and return the file
-// descriptor to the leaf directory. This function is a base for secure
-// variants of mkdir, touch and symlink.
-func secureMkPrefix(segments []string, perm os.FileMode, uid sys.UserID, gid sys.GroupID) (int, error) {
+// Secure is a helper for making filesystem operations free from certain kinds of attacks.
+type Secure struct{}
```

Comment:
I'd consider making this an interface in case it could make testing a bit easier. Otherwise a concrete type works too.

Reply:
I considered this but oped-out based on what's coming next with trespassing checks.